{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "metadata = pd.read_feather('/pollard/data/functional_genomics/encode-new/metadata.feather')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>file_accession</th>\n",
       "      <th>file_format</th>\n",
       "      <th>file_type</th>\n",
       "      <th>file_format_type</th>\n",
       "      <th>output_type</th>\n",
       "      <th>file_assembly</th>\n",
       "      <th>experiment_accession</th>\n",
       "      <th>assay</th>\n",
       "      <th>biosample_term_id</th>\n",
       "      <th>biosample_term_name</th>\n",
       "      <th>...</th>\n",
       "      <th>file_download_url</th>\n",
       "      <th>genome_annotation</th>\n",
       "      <th>platform</th>\n",
       "      <th>controlled_by</th>\n",
       "      <th>file_status</th>\n",
       "      <th>s3_uri</th>\n",
       "      <th>audit_warning</th>\n",
       "      <th>audit_internal_action</th>\n",
       "      <th>audit_not_compliant</th>\n",
       "      <th>audit_error</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENCFF537UQH</td>\n",
       "      <td>fastq</td>\n",
       "      <td>fastq</td>\n",
       "      <td>None</td>\n",
       "      <td>reads</td>\n",
       "      <td>None</td>\n",
       "      <td>ENCSR296BKT</td>\n",
       "      <td>TF ChIP-seq</td>\n",
       "      <td>UBERON:0000468</td>\n",
       "      <td>whole organism</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF537UQ...</td>\n",
       "      <td>None</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>/files/ENCFF654NML/</td>\n",
       "      <td>released</td>\n",
       "      <td>s3://encode-public/2018/05/30/9ca87f71-7d3b-4a...</td>\n",
       "      <td>mild to moderate bottlenecking, missing geneti...</td>\n",
       "      <td>experiment not submitted to GEO</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENCFF723BIE</td>\n",
       "      <td>fastq</td>\n",
       "      <td>fastq</td>\n",
       "      <td>None</td>\n",
       "      <td>reads</td>\n",
       "      <td>None</td>\n",
       "      <td>ENCSR296BKT</td>\n",
       "      <td>TF ChIP-seq</td>\n",
       "      <td>UBERON:0000468</td>\n",
       "      <td>whole organism</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF723BI...</td>\n",
       "      <td>None</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>/files/ENCFF654NML/</td>\n",
       "      <td>released</td>\n",
       "      <td>s3://encode-public/2018/05/30/4d21fe06-7b53-46...</td>\n",
       "      <td>mild to moderate bottlenecking, missing geneti...</td>\n",
       "      <td>experiment not submitted to GEO</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENCFF127LDE</td>\n",
       "      <td>bam</td>\n",
       "      <td>bam</td>\n",
       "      <td>None</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ce11</td>\n",
       "      <td>ENCSR296BKT</td>\n",
       "      <td>TF ChIP-seq</td>\n",
       "      <td>UBERON:0000468</td>\n",
       "      <td>whole organism</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF127LD...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>released</td>\n",
       "      <td>s3://encode-public/2018/05/31/f3a24a3e-ec67-4f...</td>\n",
       "      <td>mild to moderate bottlenecking, missing geneti...</td>\n",
       "      <td>experiment not submitted to GEO</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENCFF201OPY</td>\n",
       "      <td>bam</td>\n",
       "      <td>bam</td>\n",
       "      <td>None</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ce11</td>\n",
       "      <td>ENCSR296BKT</td>\n",
       "      <td>TF ChIP-seq</td>\n",
       "      <td>UBERON:0000468</td>\n",
       "      <td>whole organism</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF201OP...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>released</td>\n",
       "      <td>s3://encode-public/2018/05/31/677429a6-5568-49...</td>\n",
       "      <td>mild to moderate bottlenecking, missing geneti...</td>\n",
       "      <td>experiment not submitted to GEO</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENCFF115AUY</td>\n",
       "      <td>bigWig</td>\n",
       "      <td>bigWig</td>\n",
       "      <td>None</td>\n",
       "      <td>signal of unique reads</td>\n",
       "      <td>ce11</td>\n",
       "      <td>ENCSR296BKT</td>\n",
       "      <td>TF ChIP-seq</td>\n",
       "      <td>UBERON:0000468</td>\n",
       "      <td>whole organism</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF115AU...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>released</td>\n",
       "      <td>s3://encode-public/2018/05/31/4a36051f-06ec-41...</td>\n",
       "      <td>mild to moderate bottlenecking, missing geneti...</td>\n",
       "      <td>experiment not submitted to GEO</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 55 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  file_accession file_format file_type file_format_type  \\\n",
       "0    ENCFF537UQH       fastq     fastq             None   \n",
       "1    ENCFF723BIE       fastq     fastq             None   \n",
       "2    ENCFF127LDE         bam       bam             None   \n",
       "3    ENCFF201OPY         bam       bam             None   \n",
       "4    ENCFF115AUY      bigWig    bigWig             None   \n",
       "\n",
       "              output_type file_assembly experiment_accession        assay  \\\n",
       "0                   reads          None          ENCSR296BKT  TF ChIP-seq   \n",
       "1                   reads          None          ENCSR296BKT  TF ChIP-seq   \n",
       "2              alignments          ce11          ENCSR296BKT  TF ChIP-seq   \n",
       "3              alignments          ce11          ENCSR296BKT  TF ChIP-seq   \n",
       "4  signal of unique reads          ce11          ENCSR296BKT  TF ChIP-seq   \n",
       "\n",
       "  biosample_term_id biosample_term_name  ...  \\\n",
       "0    UBERON:0000468      whole organism  ...   \n",
       "1    UBERON:0000468      whole organism  ...   \n",
       "2    UBERON:0000468      whole organism  ...   \n",
       "3    UBERON:0000468      whole organism  ...   \n",
       "4    UBERON:0000468      whole organism  ...   \n",
       "\n",
       "                                   file_download_url genome_annotation  \\\n",
       "0  https://www.encodeproject.org/files/ENCFF537UQ...              None   \n",
       "1  https://www.encodeproject.org/files/ENCFF723BI...              None   \n",
       "2  https://www.encodeproject.org/files/ENCFF127LD...              None   \n",
       "3  https://www.encodeproject.org/files/ENCFF201OP...              None   \n",
       "4  https://www.encodeproject.org/files/ENCFF115AU...              None   \n",
       "\n",
       "              platform        controlled_by file_status  \\\n",
       "0  Illumina HiSeq 4000  /files/ENCFF654NML/    released   \n",
       "1  Illumina HiSeq 4000  /files/ENCFF654NML/    released   \n",
       "2                 None                 None    released   \n",
       "3                 None                 None    released   \n",
       "4                 None                 None    released   \n",
       "\n",
       "                                              s3_uri  \\\n",
       "0  s3://encode-public/2018/05/30/9ca87f71-7d3b-4a...   \n",
       "1  s3://encode-public/2018/05/30/4d21fe06-7b53-46...   \n",
       "2  s3://encode-public/2018/05/31/f3a24a3e-ec67-4f...   \n",
       "3  s3://encode-public/2018/05/31/677429a6-5568-49...   \n",
       "4  s3://encode-public/2018/05/31/4a36051f-06ec-41...   \n",
       "\n",
       "                                       audit_warning  \\\n",
       "0  mild to moderate bottlenecking, missing geneti...   \n",
       "1  mild to moderate bottlenecking, missing geneti...   \n",
       "2  mild to moderate bottlenecking, missing geneti...   \n",
       "3  mild to moderate bottlenecking, missing geneti...   \n",
       "4  mild to moderate bottlenecking, missing geneti...   \n",
       "\n",
       "             audit_internal_action  audit_not_compliant audit_error  \n",
       "0  experiment not submitted to GEO                 None        None  \n",
       "1  experiment not submitted to GEO                 None        None  \n",
       "2  experiment not submitted to GEO                 None        None  \n",
       "3  experiment not submitted to GEO                 None        None  \n",
       "4  experiment not submitted to GEO                 None        None  \n",
       "\n",
       "[5 rows x 55 columns]"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['TF ChIP-seq', 'scRNA-seq', 'Control ChIP-seq', 'eCLIP', 'RRBS',\n",
       "       'DNase-seq', 'CRISPR RNA-seq', 'Histone ChIP-seq',\n",
       "       'Control Mint-ChIP-seq', 'total RNA-seq', 'ATAC-seq',\n",
       "       'microRNA-seq', 'long read RNA-seq', 'Control eCLIP',\n",
       "       'shRNA RNA-seq', 'Hi-C', 'WGBS', 'Repli-seq', 'RNA microarray',\n",
       "       'CRISPRi RNA-seq', 'WGS', 'RNA-PET', 'FAIRE-seq', 'DNAme array',\n",
       "       'microRNA counts', 'small RNA-seq', 'siRNA RNA-seq', 'CAGE',\n",
       "       'RAMPAGE', 'RIP-chip', 'GM DNase-seq', 'ChIA-PET', 'icSHAPE',\n",
       "       'Repli-chip', 'Circulome-seq', 'genotyping HTS', 'RNA Bind-n-Seq',\n",
       "       'RIP-seq', 'genotyping array', 'PAS-seq', 'MNase-seq', 'MS-MS',\n",
       "       '5C', 'DNA-PET', 'iCLIP', 'TAB-seq', 'MeDIP-seq', 'MRE-seq',\n",
       "       'Switchgear', \"5' RLM RACE\", 'snATAC-seq'], dtype=object)"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata.assay.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [],
   "source": [
    "metadata_ = metadata.query(' \\\n",
    "    biosample_term_name == \"GM12878\" and \\\n",
    "    assay == \"TF ChIP-seq\" and \\\n",
    "    file_format == \"bigWig\" and \\\n",
    "    file_status == \"released\" and \\\n",
    "    output_type == \"fold change over control\" and \\\n",
    "    file_assembly == \"GRCh38\"  and\\\n",
    "    biological_replicates == \"1, 2\" and\\\n",
    "    experiment_target == \"CTCF-human\" \\\n",
    "')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18613 ENCFF578TBN CTCF-human\n",
      "158784 ENCFF279CYY CTCF-human\n",
      "304883 ENCFF852CRG CTCF-human\n"
     ]
    }
   ],
   "source": [
    "for index, row in metadata_.iterrows():\n",
    "    print(index, row['file_accession'], row['experiment_target'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>file_accession</th>\n",
       "      <th>file_format</th>\n",
       "      <th>file_type</th>\n",
       "      <th>file_format_type</th>\n",
       "      <th>output_type</th>\n",
       "      <th>file_assembly</th>\n",
       "      <th>experiment_accession</th>\n",
       "      <th>assay</th>\n",
       "      <th>biosample_term_id</th>\n",
       "      <th>biosample_term_name</th>\n",
       "      <th>...</th>\n",
       "      <th>file_download_url</th>\n",
       "      <th>genome_annotation</th>\n",
       "      <th>platform</th>\n",
       "      <th>controlled_by</th>\n",
       "      <th>file_status</th>\n",
       "      <th>s3_uri</th>\n",
       "      <th>audit_warning</th>\n",
       "      <th>audit_internal_action</th>\n",
       "      <th>audit_not_compliant</th>\n",
       "      <th>audit_error</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>18613</th>\n",
       "      <td>ENCFF578TBN</td>\n",
       "      <td>bigWig</td>\n",
       "      <td>bigWig</td>\n",
       "      <td>None</td>\n",
       "      <td>fold change over control</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>ENCSR000DZN</td>\n",
       "      <td>TF ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF578TB...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>released</td>\n",
       "      <td>s3://encode-public/2017/03/24/bec82618-499b-43...</td>\n",
       "      <td>mild to moderate bottlenecking, moderate libra...</td>\n",
       "      <td>biological replicates with identical biosample...</td>\n",
       "      <td>control insufficient read depth, insufficient ...</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>158784</th>\n",
       "      <td>ENCFF279CYY</td>\n",
       "      <td>bigWig</td>\n",
       "      <td>bigWig</td>\n",
       "      <td>None</td>\n",
       "      <td>fold change over control</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>ENCSR000AKB</td>\n",
       "      <td>TF ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF279CY...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>released</td>\n",
       "      <td>s3://encode-public/2017/01/10/1c756784-e09d-48...</td>\n",
       "      <td>low read depth, mild to moderate bottlenecking...</td>\n",
       "      <td>mismatched status</td>\n",
       "      <td>control insufficient read depth, insufficient ...</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>304883</th>\n",
       "      <td>ENCFF852CRG</td>\n",
       "      <td>bigWig</td>\n",
       "      <td>bigWig</td>\n",
       "      <td>None</td>\n",
       "      <td>fold change over control</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>ENCSR000DRZ</td>\n",
       "      <td>TF ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>...</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF852CR...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>released</td>\n",
       "      <td>s3://encode-public/2016/12/13/417c186f-b42f-45...</td>\n",
       "      <td>moderate library complexity, low read depth, m...</td>\n",
       "      <td>mismatched status</td>\n",
       "      <td>None</td>\n",
       "      <td>extremely low read depth</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows × 55 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       file_accession file_format file_type file_format_type  \\\n",
       "18613     ENCFF578TBN      bigWig    bigWig             None   \n",
       "158784    ENCFF279CYY      bigWig    bigWig             None   \n",
       "304883    ENCFF852CRG      bigWig    bigWig             None   \n",
       "\n",
       "                     output_type file_assembly experiment_accession  \\\n",
       "18613   fold change over control        GRCh38          ENCSR000DZN   \n",
       "158784  fold change over control        GRCh38          ENCSR000AKB   \n",
       "304883  fold change over control        GRCh38          ENCSR000DRZ   \n",
       "\n",
       "              assay biosample_term_id biosample_term_name  ...  \\\n",
       "18613   TF ChIP-seq       EFO:0002784             GM12878  ...   \n",
       "158784  TF ChIP-seq       EFO:0002784             GM12878  ...   \n",
       "304883  TF ChIP-seq       EFO:0002784             GM12878  ...   \n",
       "\n",
       "                                        file_download_url genome_annotation  \\\n",
       "18613   https://www.encodeproject.org/files/ENCFF578TB...              None   \n",
       "158784  https://www.encodeproject.org/files/ENCFF279CY...              None   \n",
       "304883  https://www.encodeproject.org/files/ENCFF852CR...              None   \n",
       "\n",
       "       platform controlled_by file_status  \\\n",
       "18613      None          None    released   \n",
       "158784     None          None    released   \n",
       "304883     None          None    released   \n",
       "\n",
       "                                                   s3_uri  \\\n",
       "18613   s3://encode-public/2017/03/24/bec82618-499b-43...   \n",
       "158784  s3://encode-public/2017/01/10/1c756784-e09d-48...   \n",
       "304883  s3://encode-public/2016/12/13/417c186f-b42f-45...   \n",
       "\n",
       "                                            audit_warning  \\\n",
       "18613   mild to moderate bottlenecking, moderate libra...   \n",
       "158784  low read depth, mild to moderate bottlenecking...   \n",
       "304883  moderate library complexity, low read depth, m...   \n",
       "\n",
       "                                    audit_internal_action  \\\n",
       "18613   biological replicates with identical biosample...   \n",
       "158784                                  mismatched status   \n",
       "304883                                  mismatched status   \n",
       "\n",
       "                                      audit_not_compliant  \\\n",
       "18613   control insufficient read depth, insufficient ...   \n",
       "158784  control insufficient read depth, insufficient ...   \n",
       "304883                                               None   \n",
       "\n",
       "                     audit_error  \n",
       "18613                       None  \n",
       "158784                      None  \n",
       "304883  extremely low read depth  \n",
       "\n",
       "[3 rows x 55 columns]"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [],
   "source": [
    "import wget\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'../data/bigwigs/ENCFF852CRG.bigWig'"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "wget.download(metadata_.iloc[2].file_download_url, out='../data/bigwigs')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
